molecule_hetznercloud/playbooks/create.yml

```diff
@@ -5,153 +5,121 @@
-      openssh_keypair:
-        path: "{{ ssh_path }}"
-        force: true
+      community.crypto.openssh_keypair:
+        path: "{{ ssh_key_path }}"
       register: generated_ssh_key
 
-    - name: Register the SSH key name
-      set_fact:
-        ssh_key_name: "molecule-generated-{{ 12345 | random | to_uuid }}"
-
-    - name: Register SSH key for test instance(s)
-      hcloud_ssh_key:
+    - name: Register SSH key for server(s)
+      hetzner.hcloud.hcloud_ssh_key:
         name: "{{ ssh_key_name }}"
         public_key: "{{ generated_ssh_key.public_key }}"
-        api_token: "{{ lookup('env', 'HCLOUD_TOKEN') }}"
+        labels:
+          molecule: "{{ resource_namespace }}"
         state: present
 
-    - name: Create molecule instance(s)
-      hcloud_server:
-        name: "{{ item.name }}"
-        server_type: "{{ item.server_type }}"
-        ssh_keys:
-          - "{{ ssh_key_name }}"
+    - name: Create server(s)
+      hetzner.hcloud.hcloud_server:
+        name: "{{ resource_namespace }}-{{ item.name }}"
         image: "{{ item.image }}"
+        server_type: "{{ item.server_type | default('cx11') }}"
+        ssh_keys: ["{{ ssh_key_name }}"]
         location: "{{ item.location | default(omit) }}"
         datacenter: "{{ item.datacenter | default(omit) }}"
         user_data: "{{ item.user_data | default(omit) }}"
-        api_token: "{{ lookup('env', 'HCLOUD_TOKEN') }}"
-        state: present
-      register: server
-      with_items: "{{ molecule_yml.platforms }}"
+        labels:
+          molecule: "{{ resource_namespace }}"
+        state: started
+      loop: "{{ molecule_yml.platforms }}"
+      register: servers_jobs
       async: 7200
       poll: 0
 
-    - name: Wait for instance(s) creation to complete
-      async_status:
+    - name: Wait for server(s) creation to complete
+      ansible.builtin.async_status:
         jid: "{{ item.ansible_job_id }}"
-      register: hetzner_jobs
-      until: hetzner_jobs.finished
+      loop: "{{ servers_jobs.results }}"
+      register: servers
+      until: servers is finished
       retries: 300
-      with_items: "{{ server.results }}"
 
     - name: Create volume(s)
-      hcloud_volume:
-        name: "{{ item.1.name | default(item.0.name) }}"
-        server: "{{ item.0.name }}"
-        location: "{{ item.1.location | default(omit) }}"
-        size: "{{ item.1.size | default(10) }}"
-        api_token: "{{ lookup('env', 'HCLOUD_TOKEN') }}"
-        state: "present"
-      loop: "{{ molecule_yml.platforms|subelements('volumes', skip_missing=True)}}"
-      register: volumes
-      async: 7200
-      poll: 0
-
-    - name: Wait for volume(s) creation to complete
-      async_status:
-        jid: "{{ item.ansible_job_id }}"
-      register: hetzner_volumes
-      until: hetzner_volumes.finished
-      retries: 300
-      when:
-        - volumes is defined
-        - volumes.changed
-      with_items: "{{ volumes.results }}"
+      hetzner.hcloud.hcloud_volume:
```

Comment:
Why are some things async (create server, delete server, delete volumes) while the others are not?

Reply:
Creating multiple server at once is possible, but for example attaching multiple volumes or networks to a single server might not work as we might hit a server is locked error.

Reply:
We could reorder the setup order (server, volumes, networks) => (volumes, networks, servers) to speed the setup a bit, but I would prefer to do this in a future change.